duckduckgo/Android
# Sync: Engine API and remaining triggers

Task/Issue URL: https://app.asana.com/0/0/1204239356024394/f

### Description
This PR cleans up several pieces from the Sync Engine in consequence of the updated Tech Design.
Added background sync
Added sync triggered by opening Bookmarks screen

### Steps to test this PR
Details in task

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204255299433079
